### PR TITLE
make self commented code for card: titlePannedByDelta: method

### PR DESCRIPTION
**Pull request by dingua**
i was trying to understand the code.. i have found that is not very simple to understand.. this is a try to make it more "self-commented".
i have started with "- (void)card:(CardView *)card titlePannedByDelta:(CGPoint)delta" method.
Please check it and give me your feedback.

### CODE REVIEW

File: Source/CardStackController.m [hunk 1]
```diff
@@ -662,4 +640,51 @@ - (void)moveCard:(CardView *)card
     [card pop_addAnimation:springAnimation forKey:@"frame"];
 }
 
+-(void)calculateMaximumTotalTitleBarHeightBeforeSearchAppears{
```
Comment (attila-at-hyper) [inline Source/CardStackController.m]: Could you please add a space after `-` and before `{`? Just to keep formatting consistent?

File: Source/CardStackController.m [hunk 2]
```diff
@@ -188,43 +188,21 @@ - (void)cardTitlePanDidStart:(CardView *)card {
...
-    for (NSUInteger index = 0; index < self.currentCardIndex; index++) {
-        CardView *card = [self.cards objectAtIndex:index];
-        self.maximumTotalTitleBarHeightBeforeSearchAppears += (card.titleBarHeight * card.scale - CardStackOffsetToAvoidAreaBelowTheTitleToBecomeVisible) * CardStackMaximumVisibleTitleBarProportion;
-    }
-
-    if ((card.tag != self.currentCardIndex &&
-        card.tag != self.cards.count - 1) ||
-        self.cards.count == 1) {
+    [self calculateMaximumTotalTitleBarHeightBeforeSearchAppears];
+    if ([self thisIsNotTheCorrectCard:card]) {
```
Comment (attila-at-hyper) [inline Source/CardStackController.m]: I see your point. How about this?

`BOOL panningCurrentCard = (card.tag == self.currentCardIndex);`
`BOOL panningLastCard = (card.tag == self.cards.count - 1);`
`BOOL isOnlyOneCard = (self.cards.count == 1);`
`if (!panningCurrentCard || !panningLastCard || isOnlyOneCard) {`

Probably it describes the intention the best?

File: Source/CardStackController.m [hunk 3]
```diff
@@ -188,43 +188,21 @@ - (void)cardTitlePanDidStart:(CardView *)card {
...
         return;
     }
 
-    CGFloat y = self.originalCardFrame.origin.y + delta.y;
-    if (y >= 0.0f) {
+    CGFloat possibleCardFrameY = self.originalCardFrame.origin.y + delta.y;
```
Comment (attila-at-hyper) [inline Source/CardStackController.m]: What if we call it `cardOriginYAfterPan`?

File: Source/CardStackController.m [hunk 4]
```diff
@@ -188,43 +188,21 @@ - (void)cardTitlePanDidStart:(CardView *)card {
...
-            BOOL shouldRevealSearchController = (y > self.maximumTotalTitleBarHeightBeforeSearchAppears);
-            if (shouldRevealSearchController) {
-                CGRect frame = self.searchViewController.view.frame;
-                frame.origin.y = ((y - self.maximumTotalTitleBarHeightBeforeSearchAppears) - frame.size.height);
-                if (frame.origin.y <= 0) {
-                    self.searchViewController.view.frame = frame;
-                }
-            }
+       
+        if (([self shouldConsiderOpeningSearchViewController:card]) && ([self shouldRevealSearchViewController:possibleCardFrameY])) {
```
Comment (attila-at-hyper) [inline Source/CardStackController.m]: I'd remove the `(` and `)` characters before and after the method calls as they don't make a difference.

File: Source/CardStackController.m [hunk 5]
```diff
@@ -188,43 +188,21 @@ - (void)cardTitlePanDidStart:(CardView *)card {
...
+                [self calculateSearchViewControllerFrameFrom:possibleCardFrameY];
```
Comment (attila-at-hyper) [inline Source/CardStackController.m]: What if we call it `setSearchViewControllerFrameUsing:`? It not just calculates, but also sets the frame.

**Comment by attila-at-hyper** [comment]
@dingua: thanks a lot for your PR. Please have a look at my comments and let me know your thoughts.

**Comment by dingua** [comment]
it was a try, you know better the code i think..and i think your comments are right. 
thanks.

**Comment by dingua** [comment]
@attila-at-hyper  should i make these changes or you will do it ?

**Comment by attila-at-hyper** [comment]
@dingua: it'd be great if you could make those changes. Your contribution is highly appreciated.

**Comment by dingua** [comment]
@attila-at-hyper i have applied your remarks.

**Pull request closed without merging**